graylog2-plugin-interfaces/src/main/java/org/graylog2/plugin/inputs/transports/NettyTransport.java

```diff
@@ -192,6 +193,20 @@ public void stop() {
     protected LinkedHashMap<String, Callable<? extends ChannelHandler>> getBaseChannelHandlers(final MessageInput input) {
         LinkedHashMap<String, Callable<? extends ChannelHandler>> handlerList = Maps.newLinkedHashMap();
 
+        handlerList.put("exception-logger", new Callable<ChannelHandler>() {
+            @Override
+            public ChannelHandler call() throws Exception {
+                return new SimpleChannelUpstreamHandler() {
+                    @Override
+                    public void exceptionCaught(ChannelHandlerContext ctx, ExceptionEvent e) throws Exception {
+                        log.error("Error on Input [" + input.getName() + "/" + input.getId() + "] (channel "
+                                + e.getChannel().toString() + ")", e.getCause());
```

Comment:
Would it make sense to log the inner most exception (with [`Throwables#getRootCause(java.lang.Throwable)`](http://docs.guava-libraries.googlecode.com/git/javadoc/com/google/common/base/Throwables.html#getRootCause(java.lang.Throwable)))?